src/DynamoApplications/StartupUtils.cs

```diff
@@ -0,0 +1,302 @@
+            //
+            var verbose = string.Empty;
+
+
+            var optionsSet = new OptionSet().Add("o=|O=", "OpenFilePath, Instruct Dynamo to open headless and run a dyn file at this path", o => openfilepath = o)
+            .Add("c=|C=", "CommandFilePath, Instruct Dynamo to open a commandfile and run the commands it contains at this path", c => commandFilePath = c)
+            .Add("l=|L=", "Running Dynamo under a different locale setting", l => locale = l)
+            .Add("p=|P=", "PresetFile, Instruct Dynamo to import the presets at this path into the opened .dyn", p => presetFile = p)
+            .Add("s=|S=", "PresetStateID, Instruct Dynamo to set the graph to the specified preset by name,"+
+            "this can be set to a statename or 'all', which will evaluate all states in the dyn", s => presetStateid = s)
+            .Add("v=|V=", "Verbose, Instruct Dynamo to output all evalautions it performs to an xml file at this path", v => verbose = v);
+
+            optionsSet.Parse(args);
+
+            //check for incompatabile parameters
+            if ((!string.IsNullOrEmpty(presetStateid) || (!string.IsNullOrEmpty(presetFile) || (!string.IsNullOrEmpty(verbose)))) && string.IsNullOrEmpty(openfilepath))
+            {
+                Console.WriteLine("you must supply a file to open if you want to load a preset or presetFile, or want to save an evaluation output ");
+                
+            }
+
+
+            return new CommandLineArguments
+            {
+                Locale = locale,
+                CommandFilePath = commandFilePath,
+                OpenFilePath = openfilepath,
+                PresetStateID = presetStateid,
+                PresetFilePath = presetFile,
+                Verbose = verbose,
+            };
+        }
+
+        public string Locale { get; set; }
+        public string CommandFilePath { get; set; }
+        public string OpenFilePath { get; set; }
+        public string PresetStateID { get; set; }
+        public string PresetFilePath { get; set; }
+        public string Verbose { get; set; }
+    }
+
+    public class Preloading
+    {
+        public static void PreloadShapeManager(ref string geometryFactoryPath, ref string preloaderLocation)
+        {
+            var exePath = Assembly.GetExecutingAssembly().Location;
+            var rootFolder = Path.GetDirectoryName(exePath);
+
+            var versions = new[]
+            {
+                LibraryVersion.Version219,
+                LibraryVersion.Version220,
+                LibraryVersion.Version221
+            };
+
+            var preloader = new Preloader(rootFolder, versions);
+            preloader.Preload();
+            geometryFactoryPath = preloader.GeometryFactoryPath;
+            preloaderLocation = preloader.PreloaderLocation;
+        }
+
+        public static DynamoModel MakeModel(bool CLI, IEnumerable<string> librariesToPreload)
```

Comment:
DynamoModel Constructor now takes a bool for CLI to control the loading path to attempt to load CLI without all dependencies, like the update manager etc.

same list of libraries to preload is added here to pass to path resolver constructor